src/diffusers/models/attention_processor.py

```diff
@@ -2198,6 +2213,15 @@ def __call__(
             ip_attention_probs = attn.get_attention_scores(query, ip_key, None)
             current_ip_hidden_states = torch.bmm(ip_attention_probs, ip_value)
             current_ip_hidden_states = attn.batch_to_head_dim(current_ip_hidden_states)
+            current_ip_hidden_states = current_ip_hidden_states.to(query.dtype)
+
+            if mask is not None:
+                mask_downsample = IPAdapterMaskProcessor.downsample(mask, batch_size, current_ip_hidden_states.shape[1], 
+                                                                    current_ip_hidden_states.shape[2])
+                
+                mask_downsample = mask_downsample.to(query.dtype).to(current_ip_hidden_states.device)
```

Comment:
```suggestion
                mask_downsample = mask_downsample.to(dtype=query.dtype, device=query.device)
```

Simpler and more readable?